zookeeper-server/src/main/java/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -736,6 +736,12 @@ public AddWatchRegistration(Watcher watcher, String clientPath, AddWatchMode mod
                     return watchManager.persistentWatches;
                 case PERSISTENT_RECURSIVE:
                     return watchManager.persistentRecursiveWatches;
+                case STANDARD_CHILD:
+                    return watchManager.childWatches;
+                case STANDARD_DATA:
+                    return watchManager.dataWatches;
+                case STANDARD_EXIST:
+                    return rc == 0 ? watchManager.dataWatches : watchManager.existWatches;
```

Comment:
This line of code will need to be revisited if `STANDARD_EXIST` triggers immediately per my other comment. If the node existed assume the watcher was trigger and return `null` so that it isn't stored.